code/game/objects/machinery/vending/new_marine_vendors.dm

```diff
@@ -250,7 +306,10 @@
 
 			var/type_p = L[3]
 
-			new type_p(loc)
+			var/item = new type_p(loc)
+			if(isitem(item))
+				var/obj/item/O = item
+				usr.put_in_hands(O)
```

Comment:
Personally I disagree with the putting in hand feature, would you be alright with asking the players and going by what they prefer the most?

Reply:
Good change, having to click twice for each item to vend + pickup THEN equip it is annoying.